src/System.Diagnostics.Process/src/System/Diagnostics/Process.Linux.cs

```diff
@@ -29,11 +29,11 @@ public static Process[] GetProcessesByName(string processName, string machineNam
             var processes = new List<Process>();
             foreach (int pid in ProcessManager.EnumerateProcessIds())
             {
-                Interop.procfs.ParsedStat parsedStat;
-                if (Interop.procfs.TryReadStatFile(pid, out parsedStat, reusableReader) &&
+                if (Interop.procfs.TryReadStatFile(pid, out Interop.procfs.ParsedStat parsedStat, reusableReader) &&
+                    Interop.procfs.TryReadStatusFile(pid, out Interop.procfs.ParsedStatus parsedStatus, reusableReader) &&
```

Comment:
It's probably not a deal breaker, but this adds quite a bit of extra work for each process.  Could you get some rough benchmarks so we can understand how negative an impact it actually is?  Hopefully it's barely measurable.

Reply:
I could put the `Interop.procfs.TryReadStatusFile` at the end so we only do it for processes that match the name.